### PR TITLE
cddivass(): refactor for clarity

### PR DESCRIPTION
**Pull request by WalterBright**
Move code so related code is close together, making it much easier to follow.

### CODE REVIEW

**Comment by WalterBright** [comment]
The ci/circleci Heisenbug failure: https://issues.dlang.org/show_bug.cgi?id=21084

File: src/dmd/backend/cod4.d [hunk 1]
```diff
@@ -1868,13 +1856,18 @@ void cddivass(ref CodeBuilder cdb,elem *e,regm_t *pretregs)
     retregs = mDX | mAX;
     if (op == OPmodass)
         retregs = mBX | mCX;
+
+    uint lib = uns ? CLIB.uldiv : CLIB.ldiv;
```
Comment (NilsLankila) [inline src/dmd/backend/cod4.d:1860]: `const`

File: src/dmd/backend/cod4.d [hunk 2]
```diff
@@ -1836,8 +1826,10 @@ void cddivass(ref CodeBuilder cdb,elem *e,regm_t *pretregs)
                 code_orrex(cdb.last(),rex);
             }
             getregs(cdb,mDX | mAX); // DX and AX will be destroyed
+            uint opr = uns ? 6 : 7;     // DIV/IDIV
```
Comment (NilsLankila) [inline src/dmd/backend/cod4.d:1829]: const

File: src/dmd/backend/cod4.d [hunk 3]
```diff
@@ -1745,17 +1733,19 @@ void cddivass(ref CodeBuilder cdb,elem *e,regm_t *pretregs)
...
-        isbyte = (sz == 1);               // 1 for byte operation
-        resreg = AX;                    // result register for * or /
-        if (uns)                        // if uint operation
-            opr = 6;                    // DIV
-        else                            // else signed
-            opr = 7;                    // IDIV
-        if (op == OPmodass)
-            resreg = DX;                // remainder is in DX
-
+        uint isbyte = (sz == 1);        // 1 for byte operation
```
Comment (NilsLankila) [inline src/dmd/backend/cod4.d:1747]: you can use it directly as assert condition as it's only used there

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/dmd/backend/cod4.d (hunk 3 above)
Comment (WalterBright) [reply src/dmd/backend/cod4.d:1747]: It's also used further down. I left it there because I'm not sure how to do byte operations better.

Comment (NilsLankila) [reply src/dmd/backend/cod4.d:1747]: ok I've missed that.

**Comment by NilsLankila** [comment]
Sorry my review was almost fully wronf, `lib` cannot be const.

File: src/dmd/backend/cod4.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/backend/cod4.d]: It's modified 2 lines down.

**Review by ghost — APPROVED** [review]

**Pull request merged**